cmd/start.go

```diff
@@ -106,17 +114,40 @@ var startCmd = &cobra.Command{
 			rtLogger.Warn("DEPRECATED: The --evaluator flag has been deprecated. " +
 				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
 		}
+
+		if viper.GetStringMapString(providerArgsFlagName) != nil {
+			rtLogger.Warn("DEPRECATED: The --sync-provider-args flag has been deprecated. " +
```

Comment:
I'm seeing this warning even when I don't pass `sync-provider-args`. Could you please double-check the logic? I would expect to only see this when it's actually being used.

Reply:
switched to a string check